### PR TITLE
Make compatible with Gubernator links for local tests

### PR DESCRIPTION
**Pull request by mnshaw**
For using Gubernator from local tests.
Add local as a query parameter to add the bucket in the link to the whitelist.

### CODE REVIEW

File: gubernator/view_base.py [hunk 1]
```diff
@@ -91,7 +91,9 @@ def render(self, template, context):
         template = JINJA_ENVIRONMENT.get_template(template)
         self.response.write(template.render(context))
 
-    def check_bucket(self, prefix):
+    def check_bucket(self, prefix, local=None):
+        if local:
+            BUCKET_WHITELIST.add(prefix)
```
Comment (timstclair) [inline gubernator/view_base.py]: Do the ALL_CAPS of the `BUCKET_WHITELIST` indicate that it's a constant? If so, either make a copy or change its case.

File: gubernator/templates/build.html [hunk 2]
```diff
@@ -9,7 +9,7 @@
 <script>window.onload = fix_timestamps;</script>
 <div id="header" class="container">
 <img src="/static/logo.svg">
-<h1>{% if pr %}<a href="/pr/{{pr}}">PR #{{pr}}</a> {% endif %}<a href="/builds{{job_dir}}">{{job}}</a> #{{build}}</h1>
+<h1>{% if pr %}<a href="/pr/{{pr}}">PR #{{pr}}</a> {% endif %} <a href="/builds{{job_dir|dirname}}?local={{local}}">{{job}}</a> #{{build}}</h1>
```
Comment (timstclair) [inline gubernator/templates/build.html]: Are you adding local in all these places just to pass the query parameter along (so it's not lost when navigating)? If so, I think there should be a way with Angular to install a navigation handler that preserves whitelist parameters.

**Comment by timstclair** [comment]
Have you tested a bunch with non-local builds as well?

**Comment by rmmh** [comment]
You might as well remove the bucket whitelist altogether with this feature.

What are you trying to accomplish?

**Comment by mnshaw** [comment]
I wanted people to be able to view log results from their own buckets (buckets not in whitelist), but I guess if we're doing that the whitelist really isn't necessary anymore because Gubernator would be able to access any public bucket given the local query parameter. 
I'll remove the whitelist.

**Comment by rmmh** [comment]
The whitelist exists to avoid trivial attacks that make ridiculous buckets
and force Gubernator to waste time rendering them, but that might not be a
real problem.

**Comment by mnshaw** [comment]
@rmmh  @timstclair 
Can this get merged or is removing the whitelist a legitimate concern?
If so what can be done?

**Comment by timstclair** [comment]
LGTM from my side.

**Comment by timstclair** [comment]
@rmmh Talked to some people offline, and I think we're OK with this change. Can we merge this or do you still have concerns?  (BTW - Marie's last day is Friday)

**Pull request merged**